### PR TITLE
Code Improvements in Upgrade Automation

### PR DESCRIPTION
**Pull request by jyejare**
This PR includes:
1. Code Improvements (Reduced number of code lines, made upgrade job more easier to run and rhevm info is completely hidden from user)
2. Some fixes triggered my new hammer module.
3. More detailed comments for every task for easy understanding of a code
4. Temporary workaround for using overtsdk python library version less than 4.0 as upgrade jobs failing due to massive changes in api's of ovirtsdk in 4.0 latest version. (The whole new PR will be raised for ovirtsdk 4.0 version support in our automation)
, and etc ...

### CODE REVIEW

**Comment by jyejare** [comment]
closes #393 .

**Comment by jyejare** [comment]
This PR is dependent of PR https://github.com/SatelliteQE/robottelo-ci/pull/352 and vice versa.

File: automation_tools/satellite6/upgrade/__init__.py [hunk 1]
```diff
@@ -50,13 +50,14 @@ def satellite6_upgrade(admin_password=None):
         e.g '6.1','6.2'
     """
     to_version = os.environ.get('TO_VERSION')
+    sat_host = os.environ.get('SAT_HOST')
```
Comment (sghai) [inline automation_tools/satellite6/upgrade/__init__.py]: Can we rename `sat_host` to `rhev_sat_host` just to differentiate between user provided Satellite Host and Host created from RHEVM image ?

**Comment by sghai** [comment]
ACK, pending few comments. Also please rebase this PR.

**Pull request closed without merging**